ui/gtk4.c

```diff
@@ -0,0 +1,2176 @@
+                                                 shortcut);
+        }
+        gd_update_caption(s);
+    }
+}
+
+static gboolean gtkinit;
+static void setup_actions(GtkDisplayState *ds, GtkApplicationWindow *window)
+{
+    GSimpleAction *action;
+    GVariant *default_state = g_variant_new_boolean(FALSE);
+
+    action = g_simple_action_new("quit", NULL);
+    g_signal_connect(action, "activate",
+                     G_CALLBACK(gd_menu_quit), ds);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(action));
+
+    action = g_simple_action_new("power_down", NULL);
+    g_signal_connect(action, "activate",
+                    G_CALLBACK(gd_menu_powerdown), NULL);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(action));
+
+    ds->pause_action = g_simple_action_new_stateful("pause", NULL,
+                                                    default_state);
+    g_signal_connect(ds->pause_action, "activate",
+                    G_CALLBACK(gd_menu_pause), ds);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(ds->pause_action));
+
+    action = g_simple_action_new("reset", NULL);
+    g_signal_connect(action, "activate",
+                    G_CALLBACK(gd_menu_reset), NULL);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(action));
+
+
+    ds->grab_on_hover_action = g_simple_action_new_stateful("grab-on-hover",
+                                                            NULL,
+                                                            default_state);
+    g_action_map_add_action(G_ACTION_MAP(window),
+                            G_ACTION(ds->grab_on_hover_action));
+
+    ds->grab_action = g_simple_action_new_stateful("grab-input", NULL,
+                                                   default_state);
+    g_signal_connect(ds->grab_action, "activate",
+                    G_CALLBACK(gd_menu_grab_input), ds);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(ds->grab_action));
+
+    ds->show_tabs_action = g_simple_action_new_stateful("show-tabs", NULL,
+                                                        default_state);
+    g_signal_connect(ds->show_tabs_action, "activate",
+                    G_CALLBACK(gd_menu_show_tabs), ds);
+    g_action_map_add_action(G_ACTION_MAP(window),
+                            G_ACTION(ds->show_tabs_action));
+
+    action = g_simple_action_new("untabify", NULL);
+    g_signal_connect(action, "activate",
+                    G_CALLBACK(gd_menu_untabify), ds);
+    g_action_map_add_action(G_ACTION_MAP(window), G_ACTION(action));
+
+    ds->show_menubar_action = g_simple_action_new_stateful("show-menubar", NULL,
+                                                           default_state);
+    g_signal_connect(ds->show_menubar_action, "activate",
+                    G_CALLBACK(gd_menu_show_menubar), ds);
+
+    gboolean show_menubar = !ds->opts->u.gtk.has_show_menubar
+                            || ds->opts->u.gtk.show_menubar;
```

Comment:
Through the code you left the operator in the line above in all conditionals, so here you should do the same for consistency.

```
    gboolean show_menubar = !ds->opts->u.gtk.has_show_menubar ||
                            ds->opts->u.gtk.show_menubar;
```